functions/src/update-metrics.ts

```diff
@@ -26,15 +27,25 @@ const computeInvestmentValue = (
   })
 }
 
-const computeTotalPool = (contracts: Contract[]) => {
-  return sum(contracts.map((contract) => sum(Object.values(contract.pool))))
+const computeTotalPool = (userContracts: Contract[], startTime = 0) => {
+  const periodFilteredContracts = userContracts.filter(
+    (contract) => contract.createdTime >= startTime
+  )
+  return sum(
+    periodFilteredContracts.map((contract) => sum(Object.values(contract.pool)))
+  )
 }
 
 export const updateMetricsCore = async () => {
-  const [users, contracts, bets] = await Promise.all([
+  const [users, contracts, bets, allPortfolioHistories] = await Promise.all([
     getValues<User>(firestore.collection('users')),
     getValues<Contract>(firestore.collection('contracts')),
     getValues<Bet>(firestore.collectionGroup('bets')),
+    getValues<PortfolioMetrics>(
+      firestore
+        .collectionGroup('portfolioHistory')
+        .where('timestamp', '>', Date.now() - 31 * DAY_MS) // so it includes just over a month ago
+    ),
```

Comment:
A cheaper way to do this would be to do three queries for history entries a day ago, a week ago, a month ago. E.g. filter for before specific times, sort by descending timestamp, and use limit(1).

Reply:
I thought about doing that, but I think it runs into some of the limitations of firestore. 

If you wanted to query the monthly portfolio for example, you'd want to:
 -do a collectionGroup query for 'portfolioHistory'
- where one_week_ago > timestamp > one_month_ago
- orderBy desc
- groupBy userId
- limit(1) within each groupBy group

But firestore doesn't support groupBy at all. 

If you don't add limit(1) it defeats the purpose  of the optimization - you still need to iterate. 

I'd suggest leaving it as is - by doing the timestamp check I'm already doing I ensure that the max number of values you'd have to iterate through per user is around 3k (i.e num of 15min intervals in a month).

Reply:
Ah, yeah that's true! It wouldn't be that simple. You'd have to do those three queries per user. I suppose we can punt on that for now, though I do think loading the portfolio history will be the largest cost here. Firebase charges by document loaded.

Re: iterating in reverse. Reversing modifies the array, so it would be iterating through all the elements. Plus, `[...portfolioHistory]` creates another copy of the array. I'm not as concerned about the performance though.